dcos_installer/upgrade.py

```diff
@@ -54,51 +54,62 @@
     exit 1
 fi
 
-
 # check for version of dc/os upgrading from
-version=`grep "version" /opt/mesosphere/etc/dcos-version.json | cut -d '"' -f 4`
-if [ $version != {{ installed_cluster_version }} ]; then
-   echo "ERROR: Expecting to upgrade DC/OS from {{ installed_cluster_version }} to {{ installer_version }}.\
-                Version found on node: $version"
-   exit 1
+found_version=`grep "version" /opt/mesosphere/etc/dcos-version.json | cut -d '"' -f 4`
+if [ "$found_version" != "{{ installed_cluster_version }}" ]; then
+    echo "ERROR: Expecting to upgrade DC/OS from {{ installed_cluster_version }} to {{ installer_version }}." \
+         "Version found on node: $found_version"
+    exit 1
 fi
 
-echo "Upgrading DC/OS agent {{ installed_cluster_version }}  -> {{ installer_version }}"
-
-
-pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }}
-
+# Determine this node's role.
+ROLE_DIR=/etc/mesosphere/roles
+if [ -f $ROLE_DIR/master ]; then
+    role="master"
+    role_name="master"
+elif [ -f $ROLE_DIR/slave ]; then
+    role="slave"
+    role_name="agent"
+elif [-f $ROLE_DIR/slave_public ]; then
+    role="slave_public"
+    role_name="public agent"
+else
+    echo "ERROR: Can't determine this node's role." \
+         "One of master, slave, or slave_public must be present under $ROLE_DIR."
+    exit 1
+fi
 
-pkgpanda activate --no-block {{ cluster_packages }}
+echo "Upgrading DC/OS $role_name {{ installed_cluster_version }} -> {{ installer_version }}"
+pkgpanda fetch --repository-url={{ bootstrap_url }} {{ cluster_packages }} > /dev/null
+pkgpanda activate --no-block {{ cluster_packages }} > /dev/null
```

Comment:
would it be useful to write the output of these commands to a log file somewhere just in case the output is needed for debugging? Maybe something like `/var/log/dcos/pkgpanda/upgrade_{{ installer_version }}.log`